src/memray/_memray.pyx

```diff
@@ -69,12 +69,12 @@ from libcpp.unordered_map cimport unordered_map
 from libcpp.utility cimport move
 from libcpp.vector cimport vector
 
-from ._destination import Destination
-from ._destination import FileDestination
-from ._destination import SocketDestination
-from ._metadata import Metadata
-from ._stats import Stats
-from ._thread_name_interceptor import ThreadNameInterceptor
+from memray._destination import Destination
+from memray._destination import FileDestination
+from memray._destination import SocketDestination
+from memray._metadata import Metadata
+from memray._stats import Stats
+from memray._thread_name_interceptor import ThreadNameInterceptor
```

Comment:
Something about how `scikit-build-core` does editable installs is causing this to be necessary... Things work fine with a non-editable install without these changes, but not with an editable install.

Reply:
Ah, presumably this:

> The default `editable.mode`, "redirect", uses a custom redirecting finder to combine the static CMake install dir with the original source code.

It's probably winding up wrong about what its package is due to that...

Reply:
But somehow it's working for `memray._test_utils`, which _also_ does
```py
from ._destination import Destination
```
Hm, there's still something I don't understand here...

Reply:
Ah, the thing that's throwing it off is that there's both a `memray/_memray/` directory and a `memray._memray` module, and they're not the same. I think this is a bug in `scikit-build-core`'s editable mode. I'll try to whip up an MRE to report the bug to them when I can. It should be easy to demonstrate that `__package__` changes depending on whether or not the install is editable.

Reply:
That's been fixed in scikit-build-core 0.10.6